File: sample/ScaffoldingSample.TypeScript/Models/Product.ts
import { Category } from './Category';
import { OrderDetail } from './OrderDetail';

export interface Product {
    productId: number;
    productName: string;
    categoryId: number;
    unitPrice: number;
    discontinued: boolean;
    rowVersion: any;
    category: Category;
    orderDetails: OrderDetail[];
}

File: sample/ScaffoldingSample.Embedded/Models/Product.cs
using System;
using System.Collections.Generic;

namespace ScaffoldingSample.Embedded.Models
{
    public partial class Product
    {
        public Product()
        {
            OrderDetails = new HashSet<OrderDetail>();
        }

        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public int? CategoryId { get; set; }
        public decimal? UnitPrice { get; set; }
        public bool Discontinued { get; set; }
        public byte[] RowVersion { get; set; }

        public virtual Category Category { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
    }
}
